dask/highlevelgraph.py

```diff
@@ -110,7 +111,7 @@ def get_output_keys(self) -> AbstractSet:
         keys: AbstractSet
             All output keys
         """
-        return self.keys()
+        return self.keys()  # this implementation will materialize the graph
```

Comment:
I'm not sure if this sort of commentary is frowned on or not, but I think it's an important piece of information to remind people about.

Reply:
I think that kind of thing is fine for now. Eventually, it might be nice to have a decorator that has to be included on all methods that materialize the graph, then there could be a test that makes sure that all the un-decorated methods do not materialize.